app/components/primer/select_menu_component.rb

```diff
@@ -0,0 +1,460 @@
+    #     <% end %>
+    #     <h4>No results</h4>
+    #     <p>There are no results to show.</p>
+    #   <% end %>
+    #
+    # @example 136|Loading|When fetching large lists, consider showing a loading message.
+    #   <%= render Primer::SelectMenuComponent.new(loading: true) do |component| %>
+    #     <%= component.slot(:summary, title: "Pick an item") do %>
+    #       Choose an option
+    #       <span class="dropdown-caret"></span>
+    #     <% end %>
+    #     <%= component.slot(:footer) do %>
+    #       Loading...
+    #     <% end %>
+    #   <% end %>
+    #
+    # @example 193|details-menu example|Use a details-menu instead of a div for the `.SelectMenu` element.
+    #   <%= render Primer::SelectMenuComponent.new(details_overlay: :default, reset_details: true, position: :relative, menu_tag: :"details-menu") do |component| %>
+    #     <%= component.slot(:summary) do %>
+    #       Choose an option
+    #     <% end %>
+    #     <%= component.slot(:header) do %>
+    #       My menu
+    #     <% end %>
+    #     <%= component.slot(:item, selected: true) do %>
+    #       Item 1
+    #     <% end %>
+    #     <%= component.slot(:item) do %>
+    #       Item 2
+    #     <% end %>
+    #     <%= component.slot(:item) do %>
+    #       Item 3
+    #     <% end %>
+    #   <% end %>
+    #
+    # @param align_right [Boolean] Align the whole menu to the right or not.
+    # @param loading [Boolean] Whether the content will be a loading message.
+    # @param blankslate [Boolean] Whether to style the content as a blankslate, to represent there is no content.
+    # @param list_border [Symbol] What kind of border to have around the list element. One of `:all`, `:omit_top`, or `:none`.
+    # @param message [String] A message shown above the contents.
+    # @param list_role [String] Optional `role` attribute for the list element.
+    # @param overlay [Symbol] options are `:none`, `:default`, and `:dark`. Dictates the type of overlay to render with.
+    # @param menu_tag [Symbol] HTML element type for the `.SelectMenu` tag; defaults to `:div`, could also use `:"details-menu"`.
+    # @param modal_classes [String] CSS classes to apply to the `.SelectMenu-modal` element.
+    # @param list_classes [String] CSS classes to apply to the `.SelectMenu-list` element.
+    # @param menu_classes [String] CSS classes to apply to the `.SelectMenu` element.
+    # @param message_classes [String] CSS classes to apply to the message element, if a message is included.
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    def initialize(
+      align_right: DEFAULT_ALIGN_RIGHT,
+      loading: DEFAULT_LOADING,
+      blankslate: DEFAULT_BLANKSLATE,
+      list_border: DEFAULT_LIST_BORDER_CLASS,
+      list_role: nil,
+      message: nil,
+      reset_details: false,
+      menu_tag: :div,
+      modal_classes: nil,
+      message_classes: nil,
+      menu_classes: nil,
+      list_classes: nil,
```

Comment:
In the interest of keeping this API as simple as possible, might we remove these params for now?

Reply:
Yep, can do!

Reply:
Actually, checking use cases of these, I do see these customizations would all be used in dotcom. Is it all right if they're kept?